lib/fetch/request.js

```diff
@@ -328,14 +332,9 @@ class Request {
       if (signal.aborted) {
         ac.abort()
       } else {
-        // TODO: Remove this listener on failure/success.
-        signal.addEventListener(
-          'abort',
-          function () {
-            ac.abort()
-          },
-          { once: true }
-        )
+        const abort = () => ac.abort()
+        signal.addEventListener('abort', abort, { once: true })
+        requestFinalizer.register(this, { signal, abort })
```

Comment:
We should be unregistering this after successful completion. Always passing through the registry is expensive.

Reply:
Yea, but if we could do that we wouldn't need this hack at all.